R/stock_time.R

```diff
@@ -1,51 +1,61 @@
+
+
 g3s_time_convert <- function (year, step = NULL) {
-    if (is.null(step)) {
-        as.integer(year)
-    } else if (any(step > 9)) {
-        as.integer(year) * 100L + as.integer(step)
-    } else {
-        as.integer(year) * 10L + as.integer(step)
-    }
+  if (is.null(step)) {
+    as.integer(year)
+  } else {
+    as.integer(year) * g3s_time_multiplier(year, step) + as.integer(step)
+  }
 }
 
-g3s_time_multiplier <- function (times) {
-    if (all(times > 100000)) {
-        # Year + 2-char step
-        100L
-    } else if (all(times > 10000)) {
-        # Year + step
-        10L
-    } else {
-        # Just year
-        1L
-    }
+g3s_time_multiplier <- function(year, step) {
+  g3s_year_multiplier(year) * g3s_step_multiplier(step)
+}
+
+g3s_step_multiplier <- function(step) {
+  if (is.null(step)) {
+    1L
+  } else if (all(as.integer(step) < 10)) {
+    10L
+  } else {
+    100L
+  }
+}
+
+g3s_year_multiplier <- function(year){
+  as.integer(10**(4 - min(nchar(as.integer(year)))))
 }
 
-g3s_time_labels <- function (times) {
-    mult <- g3s_time_multiplier(times)
-    if (mult > 1L) {
-        sprintf("%d-%02d",
+g3s_time_labels <- function (times, mult) {
+  if (mult > 1L) {
+    sprintf("%d-%02d",
             times %/% mult,
             times %% mult)
-    } else {
-        as.character(times)
-    }
+  } else {
+    as.character(times)
+  }
 }
 
 # Time dimension, useful for data objects
-# - times: Vector of g3s_time_convert(year, step) for each year/step data applies to
-g3s_time <- function(inner_stock, times, year = NULL, step = NULL) {
+# - time_data: dataframe that contains a year and possibly a step column
+g3s_time <- function(inner_stock, time_data, year = NULL, step = NULL) {
```

Comment:
I see why you've done it, but this interface-change breaks the symmetry between the other g3s_* functions, which all expect a list of unique values one way or another. This is leaking the work of likelihood_data into g3s_time, which I don't think should happen.

Reply:
Yep, this is back to normal with the second commit.